packages/core/src/compiler/compiler_facade_interface.ts

```diff
@@ -97,20 +91,27 @@ export enum R3FactoryTarget {
 }
 
 export interface R3DependencyMetadataFacade {
-  token: any;
-  resolved: R3ResolvedDependencyType;
+  token: unknown;
+  attribute: string|null;
   host: boolean;
   optional: boolean;
   self: boolean;
   skipSelf: boolean;
 }
 
+export interface R3DeclareDependencyMetadataFacade {
+  token?: unknown;
```

Comment:
This is required now

```suggestion
  token: unknown;
```